lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3450,10 +3450,15 @@ ParseResult FIRStmtParser::parsePropAssign() {
   auto rhsType = type_dyn_cast<PropertyType>(rhs.getType());
   if (!lhsType || !rhsType)
     return emitError(loc, "can only propassign property types");
-  if (lhsType != rhsType)
-    return emitError(loc, "cannot propassign non-equivalent type ")
-           << rhsType << " to " << lhsType;
   locationProcessor.setLoc(loc);
+  if (lhsType != rhsType) {
+    // If the lhs is anyref, and the rhs is a ClassType, insert a cast.
+    if (isa<AnyRefType>(lhsType) && isa<ClassType>(rhsType))
+      rhs = builder.create<ObjectAnyRefCastOp>(rhs);
```

Comment:
I think this is fine to be inserted regardless of IR version, right? Both AnyRefType and ClassType parsers are guarded behind the same 3.2.0 version.